src/main/java/nl/tudelft/ewi/devhub/server/database/controllers/Deliveries.java

```diff
@@ -74,15 +78,10 @@ public boolean lastDeliveryIsApprovedOrDisapproved(Assignment assignment, Group
                 .list(QDelivery.delivery);
     }
 
-    /**
-     * Get the most recent delivery for every group in this assignment
-     * @param assignment current assignment
-     * @return a list of deliveries
-     */
     @Transactional
-    public List<Delivery> getLastDeliveries(Assignment assignment) {
+    private List<Delivery> getLastDeliveriesList(Assignment assignment, Predicate expression) {
```

Comment:
I figured code duplication would be worse than a simple if, but yes it is debatable. Your call in the end ;p